src/main/scala/org/neo4j/spark/reader/Neo4jDataSourceReader.scala

```diff
@@ -19,20 +21,43 @@ class Neo4jDataSourceReader(private val options: DataSourceOptions, private val
   private val neo4jOptions: Neo4jOptions = new Neo4jOptions(options.asMap())
     .validate(options => Validations.read(options, jobId))
 
-  override def readSchema(): StructType = {
-    val schemaService = new SchemaService(neo4jOptions, jobId)
+
```

Comment:
this whole new section is quite hard to understand / reason about, can you do somethign with the naming of methods/params or add a few comments

Reply:
done